pypose/module/icp.py

```diff
@@ -0,0 +1,100 @@
+import torch
+from .. import lietensor
+from .. import knn, svdtf
+
+class ICP(torch.nn.Module):
+    r'''
+    This class implements the batched Iterative Closest Point (ICP) algorithm to find a
+    similarity transformation ( :math:`T` ) between two sets of 3-dimensional points
+    using Singular Value Decomposition (SVD). It's important to note that the solution
+    found is only a local optimum.
+
+    Args:
+        steplim (``int``, optional): The maximum number of ICP iteration steps.
+            Default: 200.
+        tol (``double``, optional): The tolerance of the relative error used to terminate
+            the algorithm. Default: 1e-6.
+        tf (``LieTensor``, optional): The initial transformation :math:`T_{init}` in
+            ``SE3type``. Default: None.
+
+    The algorithm takes two input point clouds: source point cloud (psrc) and target
+    point cloud (ptgt). The objective is to find the optimal similarity transformation
+    (:math:`T`) to minimize the error between the transformed source point cloud and the
+    target point cloud as shown in the equation:
+
+    .. math::
+        \begin{align*}
+            \underset{T}{\operatorname{arg\,min}} \sum_i \|ptgt_j - T \cdot psrc_i\|,
+        \end{align*}
```

Comment:
We don't use string such as 'ptgt' in math variable name. use normal fonts in super- and subscript, not italic. see generated doc.